### PR TITLE
test(frontend): token.services

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

To make the index canister optional in #3334 we need to test the `custom-token.services` which itself uses the `token.services`. That's why we need test for that services. Of course, we also need test generally speaking.

### CODE REVIEW

File: src/frontend/src/tests/lib/services/token.services.spec.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+		});
+
+		it('should handle asynchronous callback', () =>
+			new Promise<void>((done) => {
+				// eslint-disable-next-line require-await
+				const callback = async () => {
+					done();
+				};
+
+				loadTokenAndRun({ token: mockValidToken, callback });
```
Comment (AntonioVentilii) [inline src/frontend/src/tests/lib/services/token.services.spec.ts:43]: i am not really sure here, how is this test "tested"?

File: src/frontend/src/tests/lib/services/token.services.spec.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+		const mockErrorMessage = 'Error loading token';
+
+		let params: AutoLoadTokenParams<SaveUserToken, Erc20Token>;
+
+		let spyToastsError: MockInstance;
+		let spyBusyStart: MockInstance;
+		let spyBusyStop: MockInstance;
+
+		// we mock console.error just to avoid unnecessary logs while running the tests
+		vi.spyOn(console, 'error').mockImplementation(() => undefined);
```
Comment (AntonioVentilii) [inline src/frontend/src/tests/lib/services/token.services.spec.ts:74]: Just as note: I had to do the same in another test, maybe it is worth to create some sort of `disableConsole` util for tests

File: src/frontend/src/tests/lib/services/token.services.spec.ts (hunk 1 above)
Comment (peterpeterparker) [reply src/frontend/src/tests/lib/services/token.services.spec.ts:43]: With `done`. If it is not called, the test fails. It assert the promise of the callback is correctly handled.

File: src/frontend/src/tests/lib/services/token.services.spec.ts (hunk 2 above)
Comment (peterpeterparker) [reply src/frontend/src/tests/lib/services/token.services.spec.ts:74]: Yes, good idea!

File: src/frontend/src/tests/lib/services/token.services.spec.ts (hunk 1 above)
Comment (AntonioVentilii) [reply src/frontend/src/tests/lib/services/token.services.spec.ts:43]: ah! right!!! thank you for the explanation

**Review by AntonioVentilii — APPROVED** [review]
LGTM tks!

**Comment by peterpeterparker** [comment]
Relates to https://github.com/dfinity/oisy-wallet/issues/2883

File: src/frontend/src/tests/lib/services/token.services.spec.ts (hunk 2 above)
Comment (AntonioVentilii) [reply src/frontend/src/tests/lib/services/token.services.spec.ts:74]: Done with:

- PR https://github.com/dfinity/oisy-wallet/pull/5237

**Pull request merged**